scripts/configurator/parser/cmake_row.py

```diff
@@ -1,5 +1,19 @@
 import re
 from typing import Match, Optional
+from enum import Enum, auto
+
+class row_type(Enum):
+   EMPTY = auto()
+   SLASH_COMMENT = auto()
+   HASHTAG_COMMENT = auto()
+   VARIABLE = auto()
+   NOT_PARSABLE = auto()
+
+class var_type(Enum):
+   BOOL = "BOOL"
```

Comment:
I really HATE this. This is Python, there must be a better way. But, before getting there: the first point here is asking ourselves which is fixed as does not depend on us? I guess the CMake string types. Therefore, we have to keep them, but can change the rest (left side).

Now, why are we declaring the same types here? We're supposed to support a super set of those types and there won't be a bi-directional 1:1 mapping. Each of our types, will map to exactly one of those CMake types, but the opposite isn't true! Therefore, let's define a list of our types, using a different case, names etc. and avoid confusion with CMake types. From this code, it should be clear that those are two separate sets of types which have a UNIDIRECTIONAL mapping 1:1. In the opposite direction, the CMake type STRING will have make many our types mapped to, like INT, BASE_2_INT, etc.